opm/simulators/wells/WellHelpers.cpp

```diff
@@ -164,6 +164,13 @@ void sumDistributedWellEntries(Dune::DynamicMatrix<Scalar>& mat,
     std::copy(pos, allEntries.end(), &(vec[0]));
 }
 
+template void Opm::wellhelpers::sumDistributedWellEntries<Dune::FieldMatrix<double, 2, 2>, Dune::FieldVector<double, 2>, Dune::Communication<ompi_communicator_t*>>(Dune::FieldMatrix<double, 2, 2>&, Dune::FieldVector<double, 2>&, Dune::Communication<ompi_communicator_t*> const&);
```

Comment:
readability of this would be increased by a
```
#define INSTANTIATE_WH_I(T, Dim)  \
    template void sumDistributedWellEntries(Dune::FieldMatrix<T, Dim, Dim>&, \
                                            Dune::FieldVector<T, Dim>, \
                                            const Parallel::Communication&);

#define INSTANTIATE_WH(T) \
  INSTANTIATE_WH_I(2) \
  INSTANTIATE_WH_I(3) \
  INSTANTIATE_WH_I(4)

INSTANTIATE_WH(double)

#if FLOW_INSTANTIATE_FLOAT
INSTANTIATE_WH(float)
#endif
```
points being
1) less repetitive
2) less long lines
3) shorter since we used autodeduction for the template param

also the template parameter for Comm can be Parallel::Communication directly